src/Microsoft.DotNet.Wpf/src/WindowsBase/MS/Internal/AvTrace.cs

```diff
@@ -258,73 +259,66 @@ internal static bool IsDebuggerAttached()
         //  note: labels start at index 1, parameters start at index 0
         //
 
-        public string Trace( TraceEventType type, int eventId, string message, string[] labels, object[] parameters )
+        public string Trace(TraceEventType type, int eventId, string message, string[] labels, params Span<object> parameters)
         {
             // Don't bother building the string if this trace is going to be ignored.
 
-            if( _traceSource == null
-                || !_traceSource.Switch.ShouldTrace( type ))
-            {
+            if(_traceSource == null|| !_traceSource.Switch.ShouldTrace(type))
                 return null;
-            }
-
 
             // Compose the trace string.
 
             AvTraceBuilder traceBuilder = new AvTraceBuilder(AntiFormat(message)); // Holds the format string
-            ArrayList arrayList = new ArrayList(); // Holds the combined labels & parameters arrays.
+            List<object> combinedList = new(parameters.Length * 2); // Holds the combined labels & parameters arrays.
 
             int formatIndex = 0;
 
-            if (parameters != null && labels != null && labels.Length > 0)
+            if (!parameters.IsEmpty && labels?.Length > 0)
             {
                 int i = 1, j = 0;
-                for( ; i < labels.Length && j < parameters.Length; i++, j++ )
+                for (; i < labels.Length && j < parameters.Length; i++, j++)
                 {
                     // Append to the format string a "; {0} = '{1}'", where the index increments (e.g. the second iteration will
                     // produce {2} & {3}).
 
-                    traceBuilder.Append("; {" + (formatIndex++).ToString() + "}='{" + (formatIndex++).ToString() + "}'" );
+                    traceBuilder.Append($"; {{{formatIndex++}}}='{{{formatIndex++}}}'");
 
                     // If this parameter is null, convert to "<null>"; otherwise, when a string.format is ultimately called
                     // it produces bad results.
 
-                    if( parameters[j] == null )
+                    if(parameters[j] == null)
                     {
                         parameters[j] = "<null>";
```

Comment:
Could you check if any caller depend on the modification of `parameters` ? Because this place seems like the only place were we modify `parameters` so if no caller depend on it we could put `parameters[j]` in a variable instead (Which would also be cleaner IMO) and mutate it which would allow changing all the `Span<object>` to `ReadOnlySpan<object>` which would be more efficient and cleaner.

Reply:
Yeah, I originally didn't want to change this but all your points are true; if you're of the opinion we should do it, I'll investigate it and change accordingly.

Reply:
I couldn't find any call where the array was explicitly defined, so it shouldn't be an issue.

Also the underlying `TraceExtraMessages` should only get non-mutated slice of the array, so it shall be without issues.